src/daemon/control.rs

```diff
@@ -1103,6 +1103,20 @@ pub fn handle_rpc_messages(
 
                 response_tx.send(Ok(()))?;
             }
+            RpcMessageIn::ListSpendTxs(response_tx) => {
+                let db_path = revaultd.read().unwrap().db_file();
+
+                let mut spend_tx_map = db_list_spends(&db_path)?;
+                let mut listspend_entries = Vec::with_capacity(spend_tx_map.len());
+                for (_, (psbt, deposit_outpoints)) in spend_tx_map.drain() {
```

Comment:
nit: Do you really need the `drain()` here?
```
--- a/src/daemon/control.rs
+++ b/src/daemon/control.rs
@@ -1284,9 +1284,9 @@ pub fn handle_rpc_messages(
             RpcMessageIn::ListSpendTxs(response_tx) => {
                 let db_path = revaultd.read().unwrap().db_file();
 
-                let mut spend_tx_map = db_list_spends(&db_path)?;
+                let spend_tx_map = db_list_spends(&db_path)?;
                 let mut listspend_entries = Vec::with_capacity(spend_tx_map.len());
-                for (_, (psbt, deposit_outpoints)) in spend_tx_map.drain() {
+                for (_, (psbt, deposit_outpoints)) in spend_tx_map {
                     listspend_entries.push(ListSpendEntry {
```

Reply:
I found it somewhat more explicit, but i can change it

Reply:
Ok went for it to remove the `mut`, thanks!